jax-stuff.md

```diff
@@ -77,51 +78,52 @@ _styles: >
 
 ## How Does Parallelism Work in JAX?
 
-JAX supports two schools of thought for multi-device programming:
+JAX supports three levels of parallelism for multi-device programming:
 
-1. **Compiler, take the wheel!** Let the compiler automatically partition arrays and decide what communication to add to facilitate a given program. This lets you write a program on a single device and automatically run it on hundreds without changing anything. 
-2. **Just let me write what I mean, damnit!** While compilers are nice, they sometimes do the wrong thing and add communication you don't intend. Sometimes we want to be extremely explicit about what we're doing.
+1. **Compiler-based automatic parallelization** You program as if using a single global machine, and the compiler chooses how to shard data and partition computation (with some user-provided constraints).
+2. **Explicit sharding and automatic partitioning** You still have a global view but data shardings are
+explicit in JAX types, inspectable using `jax.typeof`.
+3. **Manual per-device programming** You have a per-device view of data
+and computation, and can communicate with explicit collectives.
 
-Correspondingly, JAX provides two APIs for each of these schools: **jit** (`jax.jit`) and **shard\_map** (`jax.experimental.shard_map.shard_map`).
+| Mode | View? | Explicit sharding? | Explicit Collectives? |
+|---|---|---|---|
+| Auto | Global | ❌ | ❌ |
+| Explicit | Global | ✅ | ❌ |
+| Manual | Per-device | ✅ | ✅ |
 
-1. `jax.jit` lets you specify the sharding of the inputs and outputs to a program (via `in_shardings` and `out_shardings`) and infers the rest using the [GSPMD](https://arxiv.org/abs/2105.04663) compiler. While it isn't perfect, it usually does a decent job at automatically scaling your program to any number of chips.
```

Comment:
Can we keep this here and add a third option with Explicit axes? I think it's important to tell people what the options are before we dive into detail.

Reply:
`jax.jit` is the option for both. It is really defined by the AxisType on your mesh. I'll reword it but won't keep it as is.